src/coreclr/jit/objectalloc.cpp

```diff
@@ -606,21 +659,32 @@ void ObjectAllocator::MarkEscapingVarsAndBuildConnGraph()
             continue;
         }
 
-        const unsigned bvIndex              = LocalToIndex(lclNum);
+        LclVarDsc* const lclDsc             = comp->lvaGetDesc(lclNum);
+        const unsigned   bvIndex            = LocalToIndex(lclNum);
         m_ConnGraphAdjacencyMatrix[bvIndex] = BitVecOps::MakeEmpty(&m_bitVecTraits);
 
-        if (comp->lvaTable[lclNum].IsAddressExposed())
+        if (lclDsc->IsAddressExposed())
         {
             JITDUMP("   V%02u is address exposed\n", lclNum);
             MarkLclVarAsEscaping(lclNum);
         }
+
+        // Parameters have unknown initial values.
+        //
+        if (lclDsc->lvIsParam)
```

Comment:
Do OSR locals need modelling here?

Reply:
We currently will only retype to TYP_BYREF in OSR methods, so we're excluding the OSR locals (and more) during the analysis.

We could model things more accurately now by giving each OSR local an unknown correction and removing the broader analysis exclusion. I'll push an update for this.